File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/crud/annotation/PersistCopyOnFetch.java
package com.antelopesystem.crudframework.crud.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Whether or not to persist a copy of the original state of the entity on fetch
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface PersistCopyOnFetch {

}

File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/crud/annotation/PersistCopyOnFetch.kt
package com.antelopesystem.crudframework.crud.annotation

/**
 * Entity annotation, whether or not to persist a copy of the original state of the entity on fetch
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class PersistCopyOnFetch 